test/errors-spec.js

```diff
@@ -0,0 +1,345 @@
+/* global beforeEach: false */
+/* global jasmine: false */
+/* global expect: false */
+/* global it: false */
+/* global describe: false */
+
+var parseTree = require("../src/parseTree");
+var Settings = require("../src/Settings");
+
+var defaultSettings = new Settings({});
+
+beforeEach(function() {
+    jasmine.addMatchers({
+        toFailWithParseError: function(util, customEqualityTesters) {
+            var prefix = "KaTeX parse error: ";
+            return {
+                compare: function(actual, expected) {
+                    try {
+                        parseTree(actual, defaultSettings);
+                        return {
+                            pass: false,
+                            message: "'" + actual + "' parsed without error"
+                        };
+                    } catch (e) {
+                        if (expected === undefined) {
+                            return {
+                                pass: true,
+                                message: "'" + actual + "' parsed with error"
+                            };
+                        }
+                        var msg = e.message;
+                        var exp = prefix + expected;
+                        if (msg === exp) {
+                            return {
+                                pass: true,
+                                message: "'" + actual + "'" +
+                                    " parsed with error '" + expected + "'"
+                            };
+                        } else if (msg.slice(0, 19) === prefix) {
+                            return {
+                                pass: false,
+                                message: "'" + actual + "'" +
+                                    " parsed with error '" + msg.slice(19) +
+                                    "' but expected '" + expected + "'"
+                            };
+                        } else {
+                            return {
+                                pass: false,
+                                message: "'" + actual + "'" +
+                                    " caused error '" + msg +
+                                    "' but expected '" + exp + "'"
+                            };
+                        }
```

Comment:
I like how provide different output depending on whether it's a parse error or not.